lib/fabrik/DockerInstance.js

```diff
@@ -48,17 +48,26 @@ class DockerInstance extends BaseInstance {
     return false;
   }
 
+  get platformContext() {
+    return this.inspectContainer()
+      .then(() => JSON.parse(this.getEnvironment().context));
+  }
+
   create(params) {
     const parameters = params.parameters;
-    const space_guid = params.space_guid;
     const exposedPorts = this.manager.exposedPorts;
-    const options = {};
+    const options = {
+      context: params.context
+    };
     return this
       .buildContainerOptions(parameters, exposedPorts, options)
       .then(opts => this.createAndStartContainer(opts, true))
       .catchThrow(DockerError.Conflict, new ServiceInstanceAlreadyExists(this.guid))
       .then(() => this.ensureContainerIsRunning(true))
-      .then(() => this.createSecurityGroup(space_guid));
+      .then(() => this.platformManager.postInstanceProvisionOperations({
+        ruleId: this.securityGroupName,
```

Comment:
atleast the securitygroup name can be directly within platform manager right? Why should this class care about this name? Move all code related to security group name to platform manager

Reply:
Removed.

Reply:
Thought of platformManager to be generic and methods would be independent. As someone might need to create a security group of any name, in that case let caller decide which name it should pass. Platform manager should be unaware of that, it will just create. similarly for delete and ensure security group methods also.